jdbc/src/java/org/apache/hive/jdbc/CsrfHttpRequestInterceptor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hive.jdbc;
+
+import org.apache.http.HttpException;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.protocol.HttpContext;
+
+import java.io.IOException;
+
+public class CsrfHttpRequestInterceptor implements HttpRequestInterceptor {
+
+  // Note : This implements HttpRequestInterceptor rather than extending
+  // HttpRequestInterceptorBase, because that class is an auth-specific
+  // class and refactoring would kludge too many things that are potentially
+  // public api.
+  //
+  // At the base, though, what we do is a very simple thing to protect
+  // against CSRF attacks, and that is to simply add another header. If
+  // HS2 is running with an CSRF filter enabled, then it will reject all
+  // requests that do not contain this. Thus, we add this in here on the
+  // client-side. This simple check prevents random other websites from
+  // redirecting a browser that has login credentials from making a
+  // request to HS2 on their behalf.
+
+  private static boolean injectHeader = true;
```

Comment:
This is a copy of already existing [XSRF Interceptor](https://github.com/apache/hive/blob/master/jdbc/src/java/org/apache/hive/jdbc/XsrfHttpRequestInterceptor.java) where we need to inject header by default for clients using JDBC as stated in [XSRF implementation](https://issues.apache.org/jira/browse/HIVE-13853) And we are using enableHeaderInjection method to set value to false for testing purposes only

Reply:
Thanks, a minor improvement on this can be methods, enableHeaderInjection makes more sense to set the value to true here, rather than setting the sent argument, If we want to keep it true than I would have two methods `enableHeaderInjection` and `disableHeaderInjection`as these are public methods, Makes more sense to me to call setter methods then sending false in enableHeaderInjections

Reply:
Agree. Renamed that function to setInjectHeader(boolean headerInjection) to make it more readable when sending false for testing purposes.